source/megengine.js

```diff
@@ -3,30 +3,96 @@
     }
 
     open(context) {
         return context.metadata('megengine-metadata.json').then((metadata) => {
-            const obj = context.open('pkl');
-            return new megengine.Model(metadata, obj);
+            switch (megengine.modelType) {
+                case 'megengine.pickle': {
+                    const obj = context.open('pkl');
+                    return new megengine.Model(metadata, obj);
+                }
+                case 'megengine.flatbuffers': {
+                    return context.require('./megengine-schema').then(() => {
+                        megengine.schema_main = flatbuffers.get('megengine').mgb.serialization.fbs;
+                        megengine.schema_v2 = megengine.schema_main.v2;
+                        megengine.schema_param = megengine.schema_main.param;
+                        megengine.schema_dtype = megengine.schema_main;
+                        let model = null;
+                        const attachments = new Map();
+
+                        const stream = context.stream;
+                        stream.skip(4);
+                        try {
+                            const reader = flatbuffers.BinaryReader.open(stream);
+                            model = megengine.schema_v2.Model.create(reader);
+                        } catch (error) {
+                            const message = error && error.message ? error.message : error.toString();
+                            throw new megengine.Error('File format is not megengine.Model (' + message.replace(/\.$/, '') + ').');
+                        }
+                        try {
+                            const archive = zip.Archive.open(stream);
+                            if (archive) {
+                                for (const entry of archive.entries) {
+                                    attachments.set(entry[0], entry[1]);
+                                }
+                            }
+                        } catch (error) {
+                            // continue regardless of error
+                        }
+                        return new megengine.Model(metadata, model);
+                    });
+                }
+                default: {
+                    throw new megengine.Error("Unsupported megengine format '" + megengine.modelType + "'.");
+                }
+            }
         });
     }
 };
 
 megengine.Model = class {
 
     constructor(metadata, obj) {
-        this._format = 'MegEngine' + (obj.dump_info && obj.dump_info.version ? ' v' + obj.dump_info.version : '');
-        this._graphs = [ new megengine.Graph(metadata, obj) ];
+        switch (megengine.modelType) {
+            case 'megengine.pickle': {
+                this._format = 'MegEngine' + (obj.dump_info && obj.dump_info.version ? ' v' + obj.dump_info.version : '');
+                break;
+            }
+            case 'megengine.flatbuffers': {
+                this._format = 'MegEngine C++ model(v2)';
```

Comment:
Is this the official format name? See above? Follow the "Xxxx v2" pattern used by all other formats.

Reply:
It's not the official name. It's my mistake. The official format name is Megengine Mge Model

Reply:
See the comment above about formats. Is there official documentation that describes all the different formats MegEngine and what the differences are. For example, [these](https://github.com/lutzroeder/netron/pull/606#issuecomment-714611995) files have a `.mge` extension as well but are not in FlatBuffers format.

Reply:
Sorry, there seems to be no such official document.